aii-core/src/main/perl/DHCP.pm

```diff
@@ -495,43 +495,56 @@ sub read_input
     return $error;
 }
 
+# update the dhcp config file and restart daemon
+sub update_and_restart 
+{
+    my $self = shift;
+
+    my $filename = $self->option('dhcpconf');
+    $self->debug(1, "AII::DHCP: going to update dhcpd configuration $filename if needed");
+    my ($error, $changed) = $self->update_dhcp_config($filename);
+    if ($error) {
+        $self->error("AII::DHCP: failed to update dhcpd configuration $filename");
+        return;
+    }
+
+    # restart dhcpd daemon
+    if ($changed) {
+        if ($self->option('norestart')) {
+            $self->verbose("AII::DHCP: config changed but norestart set!");
+        } else {
+            my $cmd = $self->option('restartcmd');
+            $self->debug(1, "AII::DHCP: restarting dhcpd daemon using cmd '$cmd'");
+            # expects an arayref
+            $self->restart_daemon([split(/\s+/, $cmd)]);
+        } 
+    } else {
+        $self->verbose("AII::DHCP: no changes to $filename: daemon not restarted");
+    }
+    return 1;
+}
+
+# return true if dhcp config need changes
```

Comment:
that is not what this method does. it only chekcs if there any nodes scheduled for changes. it's true that if no nodes are scheduled, no config change is required. but not the other way around.